### PR TITLE
feat(jax): energy model (no grad support)

### PR DESCRIPTION
**Pull request by njzjz**
Add JAX energy model without grad support. The grad support needs discussion.
Array API is not supported in this PR as it needs more effort. (JAX has more APIs than Array API)
This PR also fixes a `skip_tf` bug introduced in #3357. When no `@property` was added, `xx.skip_tf` is always cast to `True`.

## Release Notes

- **New Features**
	- Enhanced `BaseAtomicModel` and `DPAtomicModel` classes with improved array compatibility and new output definitions.
	- Introduced new classes and attributes for better model flexibility and customization.
	- Added `EnergyFittingNet` and `DOSFittingNet` for advanced fitting capabilities.
	- New functions `get_standard_model` and `get_model` for flexible model creation based on input data.
	- Added `BaseDescriptor` and `BaseFitting` classes to streamline descriptor and fitting processes.
	- Introduced `EnergyModel` class for improved atomic model handling.

- **Bug Fixes**
	- Updated serialization logic for consistency across models.

- **Tests**
	- Enhanced testing framework to support JAX operations and added methods for JAX model evaluation.

### CODE REVIEW

File: deepmd/dpmodel/atomic_model/base_atomic_model.py [hunk 1]
```diff
@@ -191,13 +195,13 @@ def forward_common_atomic(
         ret_dict = self.apply_out_stat(ret_dict, atype)
 
         # nf x nloc
-        atom_mask = ext_atom_mask[:, :nloc].astype(np.int32)
+        atom_mask = ext_atom_mask[:, :nloc].astype(xp.int32)
         if self.atom_excl is not None:
             atom_mask *= self.atom_excl.build_type_exclude_mask(atype)
 
         for kk in ret_dict.keys():
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/dpmodel/atomic_model/base_atomic_model.py:202]: Not related to this PR.

File: deepmd/dpmodel/model/make_model.py [hunk 2]
```diff
@@ -366,17 +367,18 @@ def _format_nlist(
             nnei: int,
             extra_nlist_sort: bool = False,
         ):
+            xp = array_api_compat.array_namespace(extended_coord, nlist)
             n_nf, n_nloc, n_nnei = nlist.shape
             extended_coord = extended_coord.reshape([n_nf, -1, 3])
             nall = extended_coord.shape[1]
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/dpmodel/model/make_model.py:373]: Not related.

File: deepmd/jax/model/model.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    Parameters
+    ----------
+    data : dict
+        The data to construct the model.
+    """
+    descriptor_type = data["descriptor"].pop("type")
+    data["descriptor"]["type_map"] = data["type_map"]
+    fitting_type = data["fitting_net"].pop("type")
+    data["fitting_net"]["type_map"] = data["type_map"]
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/jax/model/model.py:29]: Fixed in 3e13a338f4f3b14ce665fa1da9a07fa43a085a31

File: deepmd/jax/atomic_model/base_atomic_model.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+def base_atomic_model_set_attr(name, value):
+    if name in {"out_bias", "out_std"}:
+        value = to_jax_array(value)
+    elif name == "pair_excl":
+        if value is not None:
+            value = PairExcludeMask(value.ntypes, value.exclude_types)
+    elif name == "atom_excl":
+        if value is not None:
+            value = AtomExcludeMask(value.ntypes, value.exclude_types)
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/jax/atomic_model/base_atomic_model.py]: Fixed in 3e13a338f4f3b14ce665fa1da9a07fa43a085a31

**Review by wanghan-iapcm — APPROVED** [review]

**Review by iProzd — APPROVED** [review]

**Pull request merged**